pkg/apis/policy/v1beta1/clusterimagepolicy_types.go

```diff
@@ -27,22 +28,37 @@ import (
 //
 // +genclient
 // +genclient:nonNamespaced
-// +genclient:noStatus
-// +genreconciler:krshapedlogic=false
-
+// +genreconciler:krshapedlogic=true
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type ClusterImagePolicy struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
 
 	// Spec holds the desired state of the ClusterImagePolicy (from the client).
 	Spec ClusterImagePolicySpec `json:"spec"`
+
+	// Status represents the current state of the ClusterImagePolicy.
+	// This data may be out of date.
+	// +optional
+	Status ClusterImagePolicyStatus `json:"status,omitempty"`
 }
 
 var (
 	_ apis.Validatable   = (*ClusterImagePolicy)(nil)
 	_ apis.Defaultable   = (*ClusterImagePolicy)(nil)
 	_ kmeta.OwnerRefable = (*ClusterImagePolicy)(nil)
+	// Check that the type conforms to the duck Knative Resource shape.
+	_ duckv1.KRShaped = (*ClusterImagePolicy)(nil)
+)
+
+const (
+	// ClusterImagePolicyReady is set when the ClusterImagePolicy has been
+	// compiled into the underlying ConfigMap properly.
+	ClusterImagePolicyConditionReady = apis.ConditionReady
+
+	ClusterImagePolicyConditionKeysInlined     apis.ConditionType = "KeysInlined"
+	ClusterImagePolicyConditionPoliciesInlined apis.ConditionType = "PoliciesInlined"
+	ClusterImagePolicyConditionCMUpdated       apis.ConditionType = "ConfigMapUpdated"
```

Comment:
Do we want to add a new status to register any issue when deleting a cip https://github.com/sigstore/policy-controller/pull/533/files#diff-993c349b6a7980667de0868f0e1c1fb43da04c80bafb462c89249ab478a76a61R155 ?

Reply:
Sure. `FinalizeKind is optional, and if implemented by the reconciler will be called when the resource's deletion timestamp is set` I am not sure what means to set the deletion timestmap.